testsuite/integration/basic/src/test/java/org/jboss/as/test/integration/bc/BouncyCastleModuleTestCase.java

```diff
@@ -0,0 +1,124 @@
+        archive.addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml"); //needed to load CDI for arquillian
+        archive.addAsManifestResource(createPermissionsXmlAsset(
+                new SecurityPermission("insertProvider"),
+                new RuntimePermission("getProtectionDomain")
+        ), "permissions.xml");
+        archive.setManifest(new StringAsset(""
+                + "Manifest-Version: 1.0\n"
+                + "Dependencies: org.bouncycastle\n"));
+        return archive;
+    }
+
+    @Test
+    @OperateOnDeployment(BC_DEPLOYMENT)
+    public void testBouncyCastleProviderIsUsableThroughJceApi() throws Exception {
+
+        BouncyCastleProvider bcProvider = null;
+        try {
+            bcProvider = new BouncyCastleProvider();
+            useBouncyCastleProviderThroughJceApi(bcProvider);
+        } catch (Exception e) {
+            e.printStackTrace();
+            if (e instanceof SecurityException && e.getMessage().contains("JCE cannot authenticate the provider")) {
+                String bcLocation = (bcProvider == null)
+                        ? ""
+                        : "(" + bcProvider.getClass().getProtectionDomain().getCodeSource().getLocation() + ")";
+                throw new Exception("Packaging with BouncyCastleProvider" + bcLocation
+                        + " is probably not properly signed for JCE usage, see server log for details.", e);
+            } else {
+                throw e;
+            }
+        }
+    }
+
+    private static void useBouncyCastleProviderThroughJceApi(BouncyCastleProvider bcProvider) throws Exception {
+
+        System.out.println("Bouncy Castle JCE Provider:");
+        System.out.println("version: " + bcProvider.getVersion());
+        System.out.println("location: " + bcProvider.getClass().getProtectionDomain().getCodeSource().getLocation());
+
+        Security.addProvider(bcProvider);
+
+        KeyGenerator keygenerator = KeyGenerator.getInstance("DES");
+        Key myDesKey = keygenerator.generateKey();
+
+        Cipher desCipher;
+
+        // Create the cipher with explicit BC provider
+        desCipher = Cipher.getInstance("DES/ECB/PKCS5Padding", bcProvider.getName());
+
+        // Initialize the cipher for encryption
+        desCipher.init(Cipher.ENCRYPT_MODE, myDesKey);
+
+        // Sensitive information
+        byte[] text = "Nobody can see me".getBytes();
+        System.out.println("Text [Byte Format]: " + text);
+        System.out.println("Text: " + new String(text));
+
+        // Encrypt the text
+        byte[] textEncrypted = desCipher.doFinal(text);
+        System.out.println("Text Encryted [Byte Format]: " + textEncrypted);
+
+        // Initialize the same cipher for decryption
+        desCipher.init(Cipher.DECRYPT_MODE, myDesKey);
+
+        // Decrypt the text
+        byte[] textDecrypted = desCipher.doFinal(textEncrypted);
+        System.out.println("Text Decryted: " + new String(textDecrypted));
```

Comment:
remove all system.outs from this method or replace them with logger trace/debug calls.